std/container/array.d

```diff
@@ -318,7 +318,25 @@ if (!is(Unqual!T == bool))
                 const nbytes = mulu(newLength, T.sizeof, overflow);
                 if (overflow)
                     assert(0);
-                _payload = (cast(T*) realloc(_payload.ptr, nbytes))[0 .. newLength];
+
+                static if (hasIndirections!T)
+                {
+                    auto newPayloadPtr = cast(T*) malloc(nbytes);
+                    newPayloadPtr || assert(false, "std.container.Array.length failed to allocate memory.");
```

Comment:
This is very Perl-ish.  Personally, I'm OK with it, but AFAIK in Phobos we prefer:
````
if (newPayloadPtr)
    assert(false, ...);
````

Reply:
I can change it (but `if (!newPayloadPtr)`; but that shouldn't even be an assertion, right? Unless release builds are guaranteed to always allocate...

Reply:
I don't understand.  Doesn't calling `malloc` always allocate? Why should it ever not allocate?

Reply:
I phrased that badly. If malloc fails (out of memory), it returns null. The assertion will catch that in a debug build, but in a release build we'll crash on the following line - the assert looks like it's handling something, but since out of memory isn't a programmer error, it's the wrong tool to use.